client/src/ui/molecules/offline-status-bar/index.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+
+import { isPayingSubscriber } from "../../../utils";
+import { useOnlineStatus } from "../../../hooks";
+import { useUserData } from "../../../user-context";
+
+import "./index.scss";
+
+export function OfflineStatusBar() {
+  const user = useUserData();
+  const onlineStatus = useOnlineStatus();
+  const statusBarRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    const statusBar = statusBarRef.current;
+    const handleOnline = () => {
+      statusBar?.classList.add("is-online");
+    };
+
+    window.addEventListener("online", handleOnline);
+
+    if (statusBar) {
+      // when the fade-out animation ends, remove the class so
+      // the statusbar is hidden not just faded out.
+      statusBar.addEventListener("animationend", () => {
+        statusBar.classList.remove("is-online");
+      });
+    }
+
+    return () => {
+      window.removeEventListener("online", handleOnline);
+    };
+  }, [onlineStatus]);
+
+  if (!isPayingSubscriber(user)) {
```

Comment:
Might make sense to rather check for the service worker (For instance if in the future the offline work is made freemium or the Service worker hasn't been activated)

e.g 
`if(window?.navigator.serviceWorker?.state)`

See https://developer.allizom.org/en-US/docs/Web/API/ServiceWorkerContainer

A

Reply:
We might run into the other problem if a user is on a private window though, right? i.e. `serviceWorker` is undefined. It is easy enough to remove the conditional 😸  Let me know your thoughts.

I mean, then again, offline probably won’t work in private mode. But then again 😸 , maybe it would be nice to be notified that you are offline?

Reply:
I guess it would also be a good idea to change this messaging up a bit. For example, say a user is a paying subscriber and they loose their connection. We could do something like:

1. You are reading offline, or
2. You are currently offline but do not have offline mode enabled. [Enable offline mode here](https://link.to/settings)
3. Upsell 😸  - You are currently offline. MDN Plus allows you to keep reading even when offline! [Get MDN Plus](https://developer.mozilla.org/)

Perhaps that is something for post-launch though

Reply:
That is true. If they are in a private window this won't work at all. Happy to leave as is for now.